src/lang/cpp/internal/Parser.java

```diff
@@ -1210,21 +1223,26 @@ public int visit(IASTProblemDeclaration declaration) {
 	}
 
 	public int visit(IASTSimpleDeclaration declaration) {
-		ISourceLocation loc = getSourceLocation(declaration);
-		boolean isMacroExpansion = isMacroExpansion(declaration);
-		IList attributes = getAttributes(declaration);
+		try {
+			ISourceLocation loc = getSourceLocation(declaration);
+			boolean isMacroExpansion = isMacroExpansion(declaration);
+			IList attributes = getAttributes(declaration);
 
-		declaration.getDeclSpecifier().accept(this);
-		IConstructor declSpecifier = stack.pop();
+			declaration.getDeclSpecifier().accept(this);
+			IConstructor declSpecifier = stack.pop();
 
-		IListWriter declarators = vf.listWriter();
-		Stream.of(declaration.getDeclarators()).forEach(it -> {
-			it.accept(this);
-			declarators.append(stack.pop());
-			addDeclaredType(br.resolveBinding(it), tr.resolveType(it));
-		});
-		stack.push(builder.Declaration_simpleDeclaration(declSpecifier, declarators.done(), attributes, loc,
-				isMacroExpansion));
+			IListWriter declarators = vf.listWriter();
+			Stream.of(declaration.getDeclarators()).forEach(it -> {
+				it.accept(this);
+				declarators.append(stack.pop());
+				addDeclaredType(br.resolveBinding(it, getSourceLocation(it)), tr.resolveType(it));
+			});
+			stack.push(builder.Declaration_simpleDeclaration(declSpecifier, declarators.done(), attributes, loc,
+					isMacroExpansion));
+		}
+		catch (RuntimeException e) {
```

Comment:
Why can this happen now for this particular node only?

Reply:
I don't know.